cli/cmd/sources.go

```diff
@@ -330,6 +374,128 @@ var sourceStatusCmd = &cobra.Command{
+		objName = obj.GetName()
+		objNamespace = obj.GetNamespace()
+		sourceNamespace = sourceNamespaceFlag
+	case k8sconsts.WorkloadKindDeployment:
+		obj, err = client.Clientset.AppsV1().Deployments(sourceNamespaceFlag).Get(ctx, argName, metav1.GetOptions{})
+		objName = obj.GetName()
+		objNamespace = obj.GetNamespace()
+		sourceNamespace = sourceNamespaceFlag
+	case k8sconsts.WorkloadKindStatefulSet:
+		obj, err = client.Clientset.AppsV1().StatefulSets(sourceNamespaceFlag).Get(ctx, argName, metav1.GetOptions{})
+		objName = obj.GetName()
+		objNamespace = obj.GetNamespace()
+		sourceNamespace = sourceNamespaceFlag
+	case k8sconsts.WorkloadKindNamespace:
+		obj, err = client.Clientset.CoreV1().Namespaces().Get(ctx, argName, metav1.GetOptions{})
+		objName = obj.GetName()
+		objNamespace = obj.GetName()
+		sourceNamespace = obj.GetName()
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	var source *v1alpha1.Source
+	selector := labels.SelectorFromSet(labels.Set{
+		k8sconsts.WorkloadNameLabel:      obj.GetName(),
+		k8sconsts.WorkloadNamespaceLabel: sourceNamespace,
+		k8sconsts.WorkloadKindLabel:      string(workloadKind),
+	})
+	sources, err := client.OdigosClient.Sources(sourceNamespace).List(ctx, metav1.ListOptions{LabelSelector: selector.String()})
+	if len(sources.Items) > 0 {
+		source = &sources.Items[0]
+	} else {
+		source = &v1alpha1.Source{
+			ObjectMeta: v1.ObjectMeta{
+				GenerateName: workload.CalculateWorkloadRuntimeObjectName(objName, workloadKind),
+				Namespace:    sourceNamespace,
+			},
+			Spec: v1alpha1.SourceSpec{
+				Workload: k8sconsts.PodWorkload{
+					Kind:      workloadKind,
+					Name:      objName,
+					Namespace: objNamespace,
+				},
+			},
+		}
+
+	}
+
+	source.Spec.DisableInstrumentation = disableInstrumentation
+
+	if len(sources.Items) > 0 {
+		source, err = client.OdigosClient.Sources(sourceNamespace).Update(ctx, source, v1.UpdateOptions{})
+	} else {
+		source, err = client.OdigosClient.Sources(sourceNamespace).Create(ctx, source, v1.CreateOptions{})
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if workloadKind == k8sconsts.WorkloadKindNamespace {
+		// if toggling a namespace, check for individually instrumented workloads
+		// alert the user that these workloads won't be affected by the command
+		selector := labels.NewSelector()
+		workloadKindRequirement, err := labels.NewRequirement(k8sconsts.WorkloadKindLabel, selection.NotIn, []string{string(k8sconsts.WorkloadKindNamespace)})
+		if err != nil {
+			return nil, err
+		}
+		selector.Add(*workloadKindRequirement)
+		sources, err := client.OdigosClient.Sources(sourceNamespace).List(ctx, metav1.ListOptions{LabelSelector: selector.String()})
+		if len(sources.Items) > 0 {
+			fmt.Printf("NOTE: Configured Namespace Source, but the following Workload Sources will not be affected:\n")
+			for _, source := range sources.Items {
+				fmt.Printf("Source: %s (Workload=%s, Kind=%s)\n", source.GetName(), source.Spec.Workload.Name, source.Spec.Workload.Kind)
```

Comment:
should we show here all sources? or just those that have the opposite "disableInstrumentation" value than the one the namespace is setting? 

e.g. the ones which are truly affected by this change